### PR TITLE
Game state

### PR DESCRIPTION
**Pull request by bencbernstein**

### CODE REVIEW

File: Components/Game.js [hunk 1]
```diff
@@ -66,18 +64,27 @@ export default class Game extends React.Component {
...
   /**
-  /*  Check if all roots have been solved, fill in remaining components, and advance to next question 
+  /*  Check if all roots have been solved, fill in remaining components, and advance to next question
   **/
   checkSolution() {
     if (this.state.solvedRoots.length === this.state.wordRoots.length) {
       this.fillInRemaining()
-      this.setState({ progress: this.state.progress + 1 })
-      setTimeout(() => this.props.nextQuestion(this.state.autohintOn), 500);
+      const newProgress = this.state.progress + 1;
```
Comment (OLI9292) [inline Components/Game.js]: in JS you can also use ++ to increment by one ex. `this.state.progress++`

File: Components/Game.js [hunk 2]
```diff
@@ -66,18 +64,27 @@ export default class Game extends React.Component {
...
+        this.checkLevelComplete()
+      });
+      setTimeout(() => this.props.nextQuestion(this.state.autohintOn), 1500);
     } else {
       this.setState({choices: this.randomChoices(this.state.answerParts, this.state.allRoots) }, this.autohint)
     }
   }
 
+  checkLevelComplete() {
+    if (this.state.progress / this.state.totalWords === 1) {
```
Comment (OLI9292) [inline Components/Game.js]: to be clear progress is referring to progress in the submodule and we setting `totalWords = 10` as a placeholder?

Comment (OLI9292) [inline Components/Game.js]: this line is a little confusing why not `this.state.progress === this.state.totalWords`?

Comment (bencbernstein) [inline Components/Game.js]: lol yup that's better

File: Components/Game.js (hunk 1 above)
Comment (bencbernstein) [inline Components/Game.js]: word

File: Components/Game.js (hunk 2 above)
Comment (bencbernstein) [inline Components/Game.js]: yes 10 = placeholder

**Pull request merged**